shared/middleware/tables_api.rb

```diff
@@ -162,6 +188,13 @@ class TablesApi < Sinatra::Base
     unsupported_media_type unless request.content_type.to_s.split(';').first == 'application/json'
     unsupported_media_type unless request.content_charset.to_s.downcase == 'utf-8'
 
+    limits = TableLimits.new(@@redis, endpoint, channel_id, table_name)
+    row_count = limits.get_approximate_row_count
+    if row_count >= @@max_table_rows
+      halt 413, {}, "Too many rows, a table may have at most #{@@max_table_rows} rows"
+    end
```

Comment:
413 --> 403

In the files api https://github.com/code-dot-org/code-dot-org/blob/staging/shared/middleware/files_api.rb#L70 we use 403 Forbidden when quota is exceeded and 413 Too Large when the individual request payload is too large, so I would recommend using the same convention here. This still leaves 401 Unauthorized available in case we later want to add a status code for an authentication failure.